README.md

```diff
@@ -1,38 +1,70 @@
+- [:collision: What is NMock?](#collision-what-is-nmock)
+- [:v: Contribution](#v-contribution)
+- [:triangular\_ruler: Architecture](#triangular_ruler-architecture)
+  - [Data source](#data-source)
+  - [Repositories](#repositories)
+- [:arrow\_left::arrow\_right: Mock Import/Export file structure](#arrow_leftarrow_right-mock-importexport-file-structure)
+  - [Export](#export)
+  - [Import](#import)
+- [:bookmark\_tabs: Helpful tips](#bookmark_tabs-helpful-tips)
+
+## :collision: What is NMock?
+Ever wondered how Android developers test map features and functionalities in their projects? Do they ride their cars or bikes to do so, or do they walk in the streets? The answer is no. 
+
+There are applications in Android called Mock applications. They help us generate fake locations and trips to test map features. 
+
+**NMock** is a tool that helps you generate fake locations and trips to test the map feature in your product. It enables you to share them with others as well. We've built this application with the help of the following products and services:
+- **[Google Map SDK](https://developers.google.com/maps/documentation)**
+-  **[Kotlin Language](https://kotlinlang.org/)** 
+- **[Kotlin Coroutines](https://kotlinlang.org/docs/coroutines-overview.html)** 
+- **[Dagger-Hilt Dependency Injection](https://dagger.dev/hilt/)**. 
+
 ![NMock Banner](https://user-images.githubusercontent.com/73066290/198220049-159f1118-b181-415d-bc61-c9a0d7b1bf87.png)
 
-> Maybe you think about how map android developers test their features and functionalities? should every person that wanna test the map applications have a car and go to the streets to test? actually, the answer is No! in Android, we have some applications that we call them "Mock", and these applications help us to generate fake locations or fake trips to test our Map application and for other purposes.
+This document is a guide on how this application works.
```

Comment:
I think now we know that this README wanna explain the application in detail. so this does not necessarily need to write.

Reply:
```suggestion
```